### ISSUE TITLE
\left\{ and \left[ (and \right counterparts) do not work properly

### ISSUE THREAD

**Issue by JacquesCarette** [Compositionality/compositionality-latex-template#2]
Using either texlive 2018 on Windows, or texlive 2016 on Mac and pdflatex, without doing funky things with fonts (regardless of whether \usepackage[T1]{fontenc} is used or not, \usepackage[utf8]{inputenc} similarly). They either do not show at all, or show weirdly (\left[ as ( but too low, \right] actually as ] but also too low, by a lot). I also tried both 10pt and 11pt, no difference.

I can provide a full paper that reproduces the issue, but I don't really want to winnow things down to a small reproducible case until it seems to really be for sure due to this template.  Note that I was using another journal's template previously, and it all worked fine, which is why I suspect the template.

**Comment by n-otter** [comment]
Please provide a MWE.

**Comment by JacquesCarette** [comment]
Now that I know someone's paying attention, I will.

**Comment by JacquesCarette** [comment]
Attached is a very small tex file (gzipped to satisfy github)
[compo_bug.tex.gz](https://github.com/Compositionality/latex-template/files/3412764/compo_bug.tex.gz)
 that reproduces the problems.

**Comment by jegi** [comment]
The problem seems to be caused by MnSymbol.sty. I believe that this happens because the class file loads lmodern.sty after you've loaded MnSymbol.sty; if they were loaded the other way round, the problem would go away (according to my experiments). The class file loads lmodern.sty after all your usepackage declarations because it is added to the BeginDocument hook, rather than being done at the relevant point in the class file. So I believe that a fix is simply to load lmodern.sty on the spot; see modified compositionalityarticle-jc.cls that does that. However, I don't know why lmodern.sty was postponed to the BeginDocument hook in the first place (maybe it has to be loaded after some other packages?), so I don't know whether making this change would break other things. 
[compositionalityarticle-jc.cls.gz](https://github.com/Compositionality/latex-template/files/5062445/compositionalityarticle-jc.cls.gz)

**Comment by mroman42** [comment]
Dear Jacques,

Thank you for noticing this problem. As far as I can understand, Jeremy solved this problem in commit e24aa572f60b6acbaad140f12aeb2deafafcfe89, which is now merged. Your minimal working example seems to compile correctly now; I attach an image and the exact code I used. I will close the issue, but please feel free to reopen if you find any other problem with the current version.

Thanks again!

----

```
\documentclass[a4paper,onecolumn, superscriptaddress,10pt,accepted=2020-05-01,issue=1, volume=2, shorttitle=papers]{compositionalityarticle}

\pdfoutput=1

\usepackage[english]{babel}
\usepackage[T1]{fontenc}
\usepackage{hyperref}
\usepackage{amsmath,amssymb,amsthm}
\usepackage{tikz}
\usetikzlibrary{matrix,arrows}
\usepackage{wrapfig}
\usepackage{stmaryrd,mathtools,MnSymbol}
\usepackage{url}
\usepackage{xspace}

\pagestyle{headings}

% useful defs
\newcommand{\define}{\triangleq}
\newcommand{\rename}{\mapsto}

\newcommand{\tmop}[1]{\ensuremath{\operatorname{\mathtt{#1}}}\xspace}
\newcommand{\assign}{\coloneq}
\newcommand{\cdummy}{\cdot}

\begin{document}

\title[Theory Presentation Combinators]{Theory Presentation Combinators}

\author{Jacques Carette}
\email{carette@mcmaster.ca}
\orcid{0000-0001-8993-9804}
\homepage{http://www.cas.mcmaster.ca/~carette}
\affiliation{McMaster University, Dept. of Computing and Software, 
    Hamilton, Ontario, Canada}

\maketitle

\begin{abstract}
MRU
\end{abstract}

\section{Motivation}\label{sec:motivation}

\subsection{Renaming}

\[
\tmop{AbelianMonoid} \define\tmop{CommutativeMonoid}
 \left[\ (\cdummy)\rename +,\ e\rename 0 \right]
\]

\begin{align*}
\tmop{MtoCM} \define &
 \left[ \tmop{U}\assign \tmop{U}, 
      (\cdummy)\assign (\cdummy), 
       \tmop{e} \assign \tmop{e},
       \tmop{right\ identity} \assign \tmop{right\ identity}, \right. \\
 & \left.{}\tmop{left\ identity} \assign \tmop{left\ identity},
       \tmop{associative} \assign \tmop{associative}, \right. \\
  & \left.{}\tmop{commutative} \assign \forall x,y:U. x\cdummy y = y\cdummy x
  \right] : \tmop{Monoid} \Rightarrow \tmop{CommutativeMonoid}
\end{align*}
\noindent where we use $\Rightarrow$ to indicate that this is a construction,
and $\assign$ to mean that this is an assignment of terms of
\tmop{Monoid} to names of $\tmop{CommutativeMonoid}$.
\[ \tmop{MtoCM} \define
\left[ 
       \tmop{commutative} \assign \forall x,y:U. x\cdummy y = y\cdummy x
\right]\\
: \tmop{Monoid} \Rightarrow \tmop{CommutativeMonoid}
\]
\noindent which is easily seen to be isomorphic to the definition we started
with. 

\begin{wrapfigure}[11]{l}{0pt}
\begin{tikzpicture}[node distance=3.0cm, auto]
  \node (P) {$\left\{ V : \tmop{Type}\right\}$};
  \node (B) [right of=P] {$\left\{\ ? : \tmop{Type}\right\}$};
  \node (A) [below of=P] {$\left\{ U : \tmop{Type}\right\}$};
  \node (C) [below of=B] {$\left\{ W : \tmop{Type}\right\}$};
  \draw[->] (P) to node {$[V \rename\ ?]$} (B);
  \draw[->] (A) to node {$[U \rename B]$} (P);
  \draw[->] (A) to node [swap] {$[U \rename W]$} (C);
  \draw[->] (C) to node [swap] {$[W \rename\ ?]$} (B);
\end{tikzpicture}
\label{fig:renaming-problem}
\end{wrapfigure}

\end{document}
```

![Screenshot from 2023-05-22 15-42-45](https://github.com/Compositionality/compositionality-latex-template/assets/5337877/6dfa7552-5118-4f2c-be95-8fc07fae4620)
